### PR TITLE
[Java][native] Fix: DateTime in deepObjects is no longer serialized correctly [fix #18742]

### LINKED ISSUE #18742: [BUG][Java][native] Regression: DateTime in deepObjects is no longer serialized correctly
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

When serializing an Object of type OffsetDateTime within a query deepObject, it is no longer formatted following the ISO-8601 standard.

##### openapi-generator version

7.6.0, introduced in 6.3.0

##### OpenAPI declaration file content or url

```yaml
openapi: 3.0.1
info:
  title: title
  version: 0.0.1
paths:
  /test:
    get:
      parameters:
        - name: deepObject
          in: query
          schema:
            $ref: '#/components/schemas/DeepObject'
          explode: true
          style: deepObject
      responses:
        204:
          description: no content
components:
  schemas:
    DeepObject:
      type: object
      properties:
        dateTime:
          type: string
          format: date-time
```

##### Generation Details

All values in the deepObject are currently serialized with String.valueOf(). Before v6.3.0 this was done with ApiClient.valueToString(), which has a special handling for objects of type OffsetDateTime to ensure dateTime is serialized following RFC 3339 / ISO-8601.

v7.6.0 generates
```java
    // add `dateTime` to the URL query string
    if (getDateTime() != null) {
      joiner.add(String.format("%sdateTime%s=%s", prefix, suffix, URLEncoder.encode(String.valueOf(getDateTime()), StandardCharsets.UTF_8).replaceAll("\\+", "%20")));
    }
```

I'd propose to instead generate
```java
    // add `dateTime` to the URL query string
    if (getDateTime() != null) {
      joiner.add(String.format("%sdateTime%s=%s", prefix, suffix, URLEncoder.encode(ApiClient.valueToString(getDateTime()), StandardCharsets.UTF_8).replaceAll("\\+", "%20")));
    }
```
reusing the existing valueToString() method.

##### Steps to reproduce

Generate the clients for the above openapi.yaml:
java -jar modules/openapi-generator-cli/target/openapi-generator-cli.jar generate -i openapi.yaml -g java --library native -o out

##### Related issues/PRs

I couldn't find similar issure, but it seems to have been introduced with d269a2a09d86c461c1e586122980a5ba347dff83

##### Suggest a fix

Use the existing valueToString() method.

### PR DESCRIPTION
**Pull request by marcel-huber-ptv**
Implements proposed solution from #18742. ApiClient.valueToString is used instead of String.valueOf to ensure correct formatting of OffsetDateTime objects.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @martin-mfg (2023/08)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/Java/libraries/native/pojo.mustache [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {{#discriminator}}
 import {{invokerPackage}}.JSON;
 {{/discriminator}}
+import {{invokerPackage}}.ApiClient;
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/Java/libraries/native/pojo.mustache:4]: can we wrap it as follows?
```
{{#supportUrlQuery}}
import {{invokerPackage}}.ApiClient;
{{/supportUrlQuery}}
```

Comment (MachineUserPTV) [reply modules/openapi-generator/src/main/resources/Java/libraries/native/pojo.mustache:5]: I don't see why not. I didn't see any changes to the samples from this change.

Comment (marcel-huber-ptv) [reply modules/openapi-generator/src/main/resources/Java/libraries/native/pojo.mustache:5]: I don't see why not. I didn't see any changes to the samples from this change.

**Comment by wing328** [comment]
thanks for the fix. let's give it a try

File: modules/openapi-generator/src/main/resources/Java/libraries/native/pojo.mustache (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator/src/main/resources/Java/libraries/native/pojo.mustache:5]: it helps to avoid blank line break

**Pull request merged**